samples/sid_dut/pytest/test_functional.py

```diff
@@ -0,0 +1,158 @@
+
+MFG_HEX_PATH = os.getenv('MFG_HEX_PATH')
+
+FAST_RESPONSE_TIMEOUT = int(os.getenv('FAST_RESPONSE_TIMEOUT'))
+FLASH_MFG_TIMEOUT = int(os.getenv('FLASH_MFG_TIMEOUT'))
+DEFAULT_TIMEOUT = int(os.getenv('DEFAULT_TIMEOUT'))
+
+
+@pytest.fixture(scope='session')
+def dut(request: pytest.FixtureRequest) -> Generator[DeviceAbstract, None, None]:
+    """Return device instance."""
+    twister_harness_config: TwisterHarnessConfig = request.config.twister_harness_config  # type: ignore
+    device_config: DeviceConfig = twister_harness_config.devices[0]
+    device_type = device_config.type
+
+    device_class: Type[DeviceAbstract] = DeviceFactory.get_device(device_type)
+
+    device = device_class(device_config)
+
+    try:
+        device.connect()
+        device.generate_command()
+        device.initialize_log_files()
+        device.run_custom_script(
+            ["nrfjprog", "--snr", device_config.id.lstrip('0'), "--recover"], FLASH_MFG_TIMEOUT)
+        device.run_custom_script(
+            ["nrfjprog", "--snr", device_config.id.lstrip('0'), "--eraseall"], FLASH_MFG_TIMEOUT)
+        device.flash_and_run()
+        device.connect()
+        program_mfg_command = ["nrfjprog", "--program", MFG_HEX_PATH, "--snr",
+                               device_config.id.lstrip('0'),  "--sectorerase", "--verify", "-r"]
+        logger.debug(program_mfg_command)
+        device.run_custom_script(program_mfg_command, FLASH_MFG_TIMEOUT)
+        yield device
+    except KeyboardInterrupt:
+        pass
+    finally:  # to make sure we close all running processes after user broke execution
+        device.disconnect()
+        device.stop()
+
+
+def wait_for_message(dut: DeviceAbstract, message, timeout=DEFAULT_TIMEOUT, tally=[]):
+    time_started = time.time()
+    for line in dut.iter_stdout:
+        if line:
+            logger.debug("#: " + line)
+        if message in line:
+            logger.info(line)
+            logger.info(
+                f"detected message \"{message}\" after {time.time() - time_started:.4f}s")
+            tally.append(time.time() - time_started)
+            return True
+        if time.time() > time_started + timeout:
+            logger.error(
+                f"message {message} has not been found within {timeout}s")
+            return False
+
+
+def wait_for_prompt(dut: DeviceAbstract, prompt='uart:~$', timeout=20):
+    time_started = time.time()
+    while True:
+        dut.write(b'\n')
+        for line in dut.iter_stdout:
+            if prompt in line:
+                logger.debug('Got prompt')
+                return True
+        if time.time() > time_started + timeout:
+            logger.error(f"prompt has not been found within {timeout}s")
+            return False
+
+
+def log_output(dut: DeviceAbstract):
+    for line in dut.iter_stdout:
+        logger.debug("#: " + line)
+        if line == "":
+            return
+
+
+BLE = 1
+FSK = 2
+LORA = 3
```

Comment:
I'm not convinced to define transports in such a way. Maybe a simple dictionary of transports will be better? or a simple class?